src/test/java/duckling/support/MockServerSocket.java

```diff
@@ -0,0 +1,39 @@
+package duckling.support;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.net.SocketAddress;
+
+public class MockServerSocket extends ServerSocket {
+    private boolean isBound = false;
+
+    public MockServerSocket() throws IOException {
+    }
+
+    @Override
+    public Socket accept() {
+        return new Socket();
+    }
+
+    @Override
+    public void bind(SocketAddress address) {
+        this.isBound = true;
+    }
+
+    @Override
+    public void close() {
+        this.isBound = false;
+    }
+
+    @Override
+    public boolean isBound() {
+        return this.isBound;
+    }
+
+    @Override
+    public boolean isClosed() {
+        return !this.isBound;
+    }
+
+}
```

Comment:
Nice clean hand-rolled test double. I'd recommend considering a rename since it looks like more of a spy (there's no built-in verification here). [XUnit Test Patterns](http://xunitpatterns.com/Test%20Double%20Patterns.html) has pretty much the definitive set of namings. Also: [some interesting reading from Gary & Justin on the terminology options](https://news.ycombinator.com/item?id=12955114).

Nitpick: I recommend `<option><ctrl>l`'ing all your files to get [hopefully?] consistent whitespace - e.g. should line 38 be blank, or just deleted? (in case that shortcut doesn't work on your IntelliJ it's called something like Reformat Code (https://www.jetbrains.com/help/idea/2016.2/reformatting-source-code.html)